pandas/tests/test_index.py

```diff
@@ -297,10 +297,6 @@ def test_constructor(self):
         # arr = np.array(5.)
         # self.assertRaises(Exception, arr.view, Index)
 
-    def test_constructor_corner(self):
-        # corner case
```

Comment:
ok, this changes the API. This is not the pupose of RangeIndex though. It should have *exactly* the same API as ``Int64Index``. So you can allow a fully specified range, e.g. ``RangeIndex(0,10)`` (maybe you allow step to default), but you *cannot* just accept a scalar (I know it seems conivenent), but this is a fundamental property of the ``Index`` type. Pls revert that part of the changes. You can propose that separately (or as a note at the top of the PR), but it must be separate.